### ISSUE TITLE
VisibilityTimeout not working for Azure Functions generated with VisualStudio 

### ISSUE THREAD

**Issue by dnlmcv** [Azure/azure-webjobs-sdk#1257]
Please provide a succinct description of the issue.

#### Repro steps

Provide the steps required to reproduce the problem

1.  Configure host.json with following value:
{
  "queues": {
     "visibilityTimeout" : "00:10:00"
    }
}

2. Restart Azure Function App

3. Add a new message to a Queue

4. The new message will trigger an Azure Function to run

5. Azure function throws an error

#### Expected behavior

I expect to have the message processed again after 10 minutes

#### Actual behavior

The message is processed with 0 timeout

#### Known workarounds

None

#### Related information 

It is related either to QueueProcessor or Host configurations

**Comment by christopheranderson** [comment]
@lindydonna - related to tooling?

**Comment by MikeStall** [comment]
I think the "queues" section in config just app[lies to Azure Queues, not Service Bus Queues.

**Comment by lindydonna** [comment]
@MikeStall Good point, I missed Service Bus in the question. So, this is a feature request for Service Bus queues. Updating title.

**Comment by dnlmcv** [comment]
@lindydonna, @MikeStall  Thank you for the responses. Indeed, the "queues" section only applies to the Azure Queues, not ASB queues.

**Comment by funkycoding** [comment]
Out of interest, is this getting any attention?

**Comment by sidhuonline61** [comment]
any idea, when will this be implemented?

**Comment by markusfoss** [comment]
Will there be support for way to be able to declare visibilityTimeout or (even better) a exponential backoffpolicy? 

With a exponential backoff policy of 2 seconds we could get this behavour using 5 redeliveries:
Invocation FAIL 
-> wait 2 secs -> FAIL 
-> wait 4 secs -> FAIL
-> wait 8 secs -> FAIL
-> wait 16 secs -> FAIL
-> wait 32 secs -> FAIL (or hopefully it succeeded ;-) )

By setting a visibilityTimeout of ie 5 seconds would also work better than the immediate redelivery setup that appears to be the default setup in Azure Functions v2. When downstream services are overloaded, this setup does not scale particularly well

**Comment by patrick-steele** [comment]
My guess is that this isn't currently possible due to the sequencing of Service Bus messages, but I prefer Microsoft to comment on that.

One idea: Complete the current message and publish a new copy of it with a [scheduled time in the future](https://docs.microsoft.com/en-us/azure/service-bus-messaging/message-sequencing#scheduled-messages).  Not quite as easy as [updating the visibility timeout](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.storage.queue.cloudqueue.updatemessage?view=azure-dotnet-legacy) like we can in storage queues, but it should work.
